evap/staff/tests/test_json_importer.py

```diff
@@ -0,0 +1,263 @@
+            "lecturers": [{"gguid": "0x3"}],
+            "students": [{"gguid": "0x1"}, {"gguid": "0x2"}],
+        },
+        {
+            "gguid": "0x6",
+            "lvnr": 2,
+            "title": "Prozessorientierte Informationssysteme",
+            "title_en": "Process-oriented information systems",
+            "type": "Klausur",
+            "isexam": True,
+            "courses": [
+                {"cprid": "BA-Inf", "scale": ""},
+                {"cprid": "MA-Inf", "scale": ""},
+            ],
+            "relatedevents": {"gguid": "0x5"},
+            "appointments": [{"begin": "29.07.2024 10:15", "end": "29.07.2024 11:45"}],
+            "lecturers": [{"gguid": "0x3"}, {"gguid": "0x4"}],
+            "students": [{"gguid": "0x1"}, {"gguid": "0x2"}],
+        },
+    ],
+}
+EXAMPLE_JSON = json.dumps(EXAMPLE_DATA)
+
+
+class TestImportUserProfiles(TestCase):
+    def setUp(self):
+        self.students = EXAMPLE_DATA["students"]
+        self.lecturers = EXAMPLE_DATA["lecturers"]
+
+        self.semester = baker.make(Semester)
+
+    def test_import_students(self):
+        self.assertEqual(UserProfile.objects.all().count(), 0)
+
+        importer = JSONImporter(self.semester)
+        importer._import_students(self.students)
+
+        user_profiles = UserProfile.objects.all()
+        self.assertEqual(user_profiles.count(), 2)
+
+        for i, user_profile in enumerate(user_profiles):
+            self.assertEqual(user_profile.email, self.students[i]["email"])
+            self.assertEqual(user_profile.last_name, self.students[i]["name"])
+            self.assertEqual(user_profile.first_name_given, self.students[i]["christianname"])
+
+        self.assertEqual(importer.statistics.name_changes, [])
+
+    def test_import_existing_students(self):
+        user_profile = baker.make(
+            UserProfile, email=self.students[0]["email"], last_name="Doe", first_name_given="Jane"
+        )
+
+        importer = JSONImporter(self.semester)
+        importer._import_students(self.students)
+
+        self.assertEqual(UserProfile.objects.all().count(), 2)
+
+        user_profile.refresh_from_db()
+
+        self.assertEqual(user_profile.email, self.students[0]["email"])
+        self.assertEqual(user_profile.last_name, self.students[0]["name"])
+        self.assertEqual(user_profile.first_name_given, self.students[0]["christianname"])
+
+        self.assertEqual(
+            importer.statistics.name_changes,
+            [
+                NameChange(
+                    old_last_name="Doe",
+                    old_first_name_given="Jane",
+                    new_last_name=self.students[0]["name"],
+                    new_first_name_given=self.students[0]["christianname"],
+                )
+            ],
+        )
+
+    def test_import_lecturers(self):
+        self.assertEqual(UserProfile.objects.all().count(), 0)
+
+        importer = JSONImporter(self.semester)
+        importer._import_lecturers(self.lecturers)
```

Comment:
In theory, you could always just inline these. I don't know if it's more readable though.

```suggestion
        JSONImporter(self.semester)._import_lecturers(self.lecturers)
```